Realm/RLMUser.mm

```diff
@@ -132,6 +141,15 @@ - (void)invalidate {
     return path;
 }
 
+- (std::string)pathForFlexibleSync {
+    if (!_user) {
+        return "";
```

Comment:
this should really be returning an optional string

Reply:
Or this can be an exception, as this is arguably an exceptional state.